### DISCUSSION TITLE
WCS VOM UK data

### DISCUSSION THREAD

**Discussion by jorgesainzdeaja** [eblondel/ows4R#121] [OGC WCS] [answered]
Hi!,

have a similar problem here, I have tried the code above and works perfectly, but when trying to get VOM data from an UK WCS service I am getting the following error:

```
Error in if (axisLatIdx < axisLonIdx) bbox <- rbind(bbox[2, ], bbox[1,  : 
  argument is of length zero
```
I was working with the following BBOX in 27700 crs:

focus_bbox_27700
    xmin     ymin     xmax     ymax 
519713.3 175841.0 522783.3 178911.0 

```
VOM <- WCSClient$new("https://environment.data.gov.uk/spatialdata/vegetation-object-model/wcs", "2.0.1", logger = "INFO")

VOM_cov <- VOM$capabilities$findCoverageSummaryById("ecae3bef-1e1d-4051-887b-9dc613c928ec__Vegetation_Object_Model_Elevation_2022", exact = F)

VOM_data <- VOM_cov$getCoverage(bbox = OWSUtils$toBBOX(focus_bbox_27700[1], focus_bbox_27700[3], focus_bbox_27700[2], focus_bbox_27700[4]))
```
What I am doing wrong??, tested to load the WCS layer on QGIS and it loads just right

**Comment by eblondel** [comment, marked as answer]
hi @jorgesainzdeaja I've tested your data source. As follow-up I made 2 enhancements (#122, #123) and everything works as expected now:
- #122 is to better handle service exceptions (in your case if you try to get the full coverage). It appears that this WCS service is not handling status code 400 when there is an exception, but a valid 200 response. 
- #123 actually targets the error you got, for which some envelope axisLabels (E N) were not managed as valid lon/lat axisLabels

Code I've tested:

```r
VOM <- WCSClient$new("https://environment.data.gov.uk/spatialdata/vegetation-object-model/wcs", "2.0.1", logger = "INFO")
VOM_cov <- VOM$capabilities$findCoverageSummaryById("ecae3bef-1e1d-4051-887b-9dc613c928ec__Vegetation_Object_Model_Elevation_2022", exact = F)
VOM_data <- VOM_cov$getCoverage(bbox = OWSUtils$toBBOX(150000,160000,656000,657000))
```

**Reply by Jorgesainzdeajacurbelo** [reply]
many thanks for this, it works perfectly!
